growattServer/__init__.py

```diff
@@ -128,106 +132,439 @@ def login(self, username, password, is_password_hashed=False):
             'type': 1,
             'date': date_str
         })
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to retrieve inverter data')
 
     def inverter_detail(self, inverter_id):
         """
-        Get "All parameters" from PV inverter.
+        Get detailed data from PV inverter.
+
+        Args:
+            inverter_id (str): The ID of the inverter.
+
+        Returns:
+            dict: A dictionary containing the inverter details.
+
+        Raises:
+            Exception: If the request to the server fails.
         """
-        response = self.session.get(self.get_url('newInverterAPI.do'), params={
+        response = self.session.get(self.__get_url('newInverterAPI.do'), params={
             'op': 'getInverterDetailData',
             'inverterId': inverter_id
         })
 
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to retrieve inverter details')
 
     def inverter_detail_two(self, inverter_id):
         """
-        Get "All parameters" from PV inverter.
+        Get detailed data from PV inverter (alternative endpoint).
+
+        Args:
+            inverter_id (str): The ID of the inverter.
+
+        Returns:
+            dict: A dictionary containing the inverter details.
+
+        Raises:
+            Exception: If the request to the server fails.
         """
-        response = self.session.get(self.get_url('newInverterAPI.do'), params={
+        response = self.session.get(self.__get_url('newInverterAPI.do'), params={
             'op': 'getInverterDetailData_two',
             'inverterId': inverter_id
         })
 
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to retrieve inverter details')
+
+    def tlx_get_system_status(self, plant_id, tlx_id):
+        """
+        Get status of the system
+
+        Args:
+            plant_id (str): The ID of the plant.
+            tlx_id (str): The ID of the TLX inverter.
+
+        Returns:
+            dict: A dictionary containing system status.
+
+        Raises:
+            Exception: If the request to the server fails.
+        """
+        response = self.session.post(
+            self.__get_url("newTlxApi.do"),
+            params={"op": "getSystemStatus_KW"},
+            data={"plantId": plant_id,
+                  "id": tlx_id}
+        )
+
+        if response.status_code == 200:
+            return response.json()['obj']
+        else:
+            raise Exception("Failed to retrieve system status")
+
+    def tlx_get_energy_overview(self, plant_id, tlx_id):
```

Comment:
```suggestion
    def tlx_energy_overview(self, plant_id, tlx_id):
```

Pulls it inline with 
https://github.com/indykoning/PyPi_GrowattServer/blob/75b6b73d3aa7f4014ce6759a3067d5c1375e74f3/growattServer/__init__.py#L495